front/src/components/oauth2/actions.js

```diff
@@ -0,0 +1,109 @@
+import { RequestStatus } from '../../utils/consts';
+import { OAUTH2 } from '../../../../server/utils/constants';
+
+const actions = store => ({
+  updateIntegrationName(state, e) {
+    store.setState({
+      integrationName: e
+    });
+  },
+  updateClientId(state, e) {
+    store.setState({
+      clientId: e.target.value
+    });
+  },
+  updateSecret(state, e) {
+    store.setState({
+      secret: e.target.value
+    });
+  },
+  async getCurrentConfig(state) {
+    store.setState({
+      oauth2GetStatus: RequestStatus.Getting
+    });
+
+    const returnServiceId = (await state.httpClient.get(`/api/v1/service/${state.integrationName}`)).id;
+
+    const returnGetConfig = await state.httpClient.get('/api/v1/service/oauth2/client', {
+      service_id: returnServiceId
+    });
+    if (returnGetConfig) {
+      store.setState({
+        clientIdInDb: returnGetConfig.client_id,
+        oauth2ErrorMsg: null,
+        oauth2GetStatus: RequestStatus.Success
+      });
+    } else {
+      store.setState({
+        oauth2ErrorMsg: null,
+        oauth2GetStatus: RequestStatus.Success
```

Comment:
Yes, in case of returnGetConfig exist the component OAuth2Config.jsx display page with information (explain integration is configured) and if not display form to set client and secret id